### ISSUE TITLE
Connecting from ACI pod to cluster service or from cluster services to ACI pod communication does not appear to work

### ISSUE THREAD

**Issue by JockDaRock** [Azure/aci-connector-k8s#51]
When I spin up an ACI pod through aci-connector I am not able to connect back to services that are on my k8s cluster (currently using GKE) or connect to my aci-pod from other services.

ACI-pod attempts to use cluster-IP addresses from what I can tell.

Is this expected behavior?

Kind Regards

**Comment by lachie83** [comment]
Yes this is expected. Services cannot currently route or lookup services via service discovery. We're hoping to make some of that work soon

**Comment by JockDaRock** [comment]
Do you have an ETA on when you might have that working?

Also, what if I deployed my entire application within ACI. Would I get the same result between service / pods?

**Comment by lachie83** [comment]
No ETA currently and yes you would get the same result between if the whole deployment was on ACI

**Comment by JockDaRock** [comment]
Any updates on this?

**Comment by feiskyer** [comment]
@JockDaRock Related upstream work is ongoing, see https://github.com/kubernetes/kubernetes/pull/53656 and https://github.com/kubernetes/kubernetes/issues/54743.

**Comment by rtyler** [comment]
The linked upstream work seems to be assigned to the 1.8 milestone for Kubernetes. Based on my read, I'm assuming these would apply to the K8s environment the connector is deployed within, and not relevant to the Azure Container Instance backend itself?

I'm also wondering if that 1.8 milestone means that this support might exist in 1.8.1, which is available in AKS right now 🙏

**Comment by rtyler** [comment]
@JockDaRock All I can say is that the aci-connector works as currently defined in AKS with 1.8.1, I haven't done the testing necessary to determine whether this ticket has been resolved.

**Comment by JockDaRock** [comment]
ok fair enough.  I will try to test today or tomorrow if I have time.  I am a little excited about the possibility of it being resolved so hopefully it is not misplaced excitement ;)
